### PR TITLE
Implemented sequential chains to generate the sections separately

### PR DESCRIPTION
**Pull request by AmgadHasan**
Implemented sectional chains that are responsible for generating specific sections of the SRS document. 
Also implemented DocGen class to link these chains and allow seamless flow of  conversation with the user (@Elghandour-eng did 90% of the work here)

### CODE REVIEW

**Comment by AmgadHasan** [comment]
Thanks! We'll merge this.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+install:
+	pip install -r requirements.txt
+run:
+	gradio run app.py
```
Comment (YasmeenNour) [inline Makefile:4]: essential part of the project

File: utils/agents.py
Comment (YasmeenNour) [inline utils/agents.py:1]: Well-organized and modularized section of code

File: utils/api.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ def get_chat_text_models(platform):
     vertexai.init(project=project_id, credentials=credentials)
     chat_model = ChatVertexAI(model_name='chat-bison', max_output_tokens=128)
     text_model = VertexAI(model_name='text-bison', max_output_tokens=1024)
+    print("text_model", text_model)
```
Comment (YasmeenNour) [inline utils/api.py:15]: What's the goal of printing the model here?

File: utils/objects.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from pydantic import BaseModel, Field, validator
+from typing import List
+
+
+class Introduction(BaseModel):
+    app_name: str = Field(description="app name of mobile application")
+    purpose: str = Field(description="purpose of mobile application")
+    scope: str = Field(description="scope of mobile app")
+    target_user: str = Field(description="target user of mobile app")
+
```
Comment (YasmeenNour) [inline utils/objects.py:11]: This sectioning added strong and valid details to generate the SRS

**Review by YasmeenNour — APPROVED** [review]
Well-organized code

File: utils/objects.py (hunk 3 above)
Comment (AmgadHasan) [reply utils/objects.py:11]: > This sectioning added strong and valid details to generate the SRS

Thanks!

File: utils/api.py (hunk 2 above)
Comment (AmgadHasan) [reply utils/api.py:15]: For debugging purposes. I was too lazy to use debugging.debug lmao. Will fix it in next push!

Comment (AmgadHasan) [reply utils/api.py:15]: > What's the goal of printing the model here?

For debugging purposes. I was too lazy to use debugging.debug lmao. 
Will fix it in next push!

**Pull request merged**